evalml/automl/automl_search.py

```diff
@@ -423,11 +430,18 @@ def __init__(self,
             self._engine = engine
 
         self.automl_config = AutoMLConfig(self.data_splitter, self.problem_type,
-                                          self.objective, self.additional_objectives, self.optimize_thresholds,
+                                          self.objective, self.additional_objectives, self.thresholding_objective, self.optimize_thresholds,
                                           self.error_callback, self.random_seed,
                                           self.X_train.ww.schema,
                                           self.y_train.ww.schema)
-
+        self.threshold_automl_config = self.automl_config
+        if is_binary(self.problem_type) and self.optimize_thresholds and self.objective.score_needs_proba:
+            # use the thresholding_objective
+            self.threshold_automl_config = AutoMLConfig(self.data_splitter, self.problem_type,
```

Comment:
This allows us to use the `thresholding_objective` as the objective to train the best pipeline with if we allow it.